### ISSUE TITLE
Implement Java-array .new(length) method

### ISSUE THREAD

**Issue by felixvf** [mirah/mirah#449]
For example

```mirah

byte[].new(6)

```

should be valid Mirah code and should create a byte[] array with a length of 6.

**Comment by uujava** [comment]
Currently mirah uses brackets as array constructor:
bytes = byte[6] # 6 element byte array
Here is macro implemented for array initializer: https://github.com/uujava/mirah/commit/1556c562d7a176ef828c4942923c3cec2d8eb055

Actually, it should be implemented differently:
```byte[6]``` should create single element array, and ```byte[].new(6)``` - 6 element empty array
But where were a lot of legacy code in test and other places so here we are.

**Comment by felixvf** [comment]
In my view,

```mirah
byte[].new(6)
```

should create a 1-dimensional array of bytes with 6 elements (and zero values).

```mirah
byte[][].new(6,9)
```

should create a 1-dimensional array of byte-arrays with 6 elements, each containing a 1-dimensional array of bytes with 9 elements (and zero values).

```mirah
[1,5,4,2].as(byte[])
```

should create a 1-dimensional array of bytes with 4 elements (and the appropriate values).


```mirah
byte[][].new(6) do |i|
  [i,i*2,i*4].as(byte[])
end
```

should create a 1-dimensional array of byte-arrays with 4 elements, each containing a 1-dimensional array of bytes with 3 elements (and the values [0,0,0],[1,2,4],[2,4,8],[3,6,12]).

```mirah
Array.new(6) do |i|
  [i,i*2,i*4].as(byte[])
end
```

should create a 1-dimensional java.util.Array of byte-arrays with 4 elements, each containing a 1-dimensional array of bytes with 3 elements (and the values [0,0,0],[1,2,4],[2,4,8],[3,6,12]).


Unfortunately, the current legacy Mirah dialect and @uujava’s Mirah dialect are conflicting regarding the meaning of `byte[6]` (a byte array of zeroes with length 6? or a byte array containing a 6 with length 1?). That this has been possible at all is a sign of that expression not being intuitively understood. Maybe it would be good to deprecate and phase out `byte[variable]` expressions altogether.

**Comment by uujava** [comment]
We work with arrays mostly in performance or memory critical code.
This ``` [1,5,4,2].as(byte[])``` has bad metrics in both memory and performance area.

**Comment by felixvf** [comment]
> We work with arrays mostly in performance or memory critical code.
> This `[1,5,4,2].as(byte[])` has bad metrics in both memory and performance area.

That’s not true. Note that `ArrayList#as` is a macro which does the equivalent of

```mirah
byte[].new(4) tap do |b|
  b[0] = 1.as!(byte)
  b[1] = 5.as!(byte)
  b[2] = 4.as!(byte)
  b[3] = 2.as!(byte)
end
```

(note that `as!(byte)` is a no-operation at the JVM level)
and not, as you might have believed,

```mirah
begin
  a = ArrayList.new(4)
  a.add(Integer.new(1))
  a.add(Integer.new(5))
  a.add(Integer.new(4))
  a.add(Integer.new(2))
  byte[].new(a.length).tap do |b|
    b.length.times do |i|
      b[i] = a[i].as!(Integer).intValue.as!(byte)
    end
  end
end
```

Thus, we can be perfectly efficient here.

**Comment by uujava** [comment]
Oh, that's nice mirah macros stuff. Sorry for misunderstanding.

**Comment by felixvf** [comment]
Fixed by #450.
